src/network/networkd-ipv6-proxy-ndp.c

```diff
@@ -0,0 +1,209 @@
+/***
+  This file is part of systemd.
+
+  Copyright 2017 Florian Klink <flokli@flokli.de>
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <netinet/ether.h>
+#include <linux/if.h>
+#include <unistd.h>
+
+#include "string-util.h"
+#include "fileio-label.h"
+#include "netlink-util.h"
+#include "networkd-link.h"
+#include "networkd-network.h"
+#include "networkd-ipv6-proxy-ndp.h"
+#include "networkd-manager.h"
+
+bool ipv6_proxy_ndp_is_enabled(Link *link) {
```

Comment:
Hm, this is pretty hard to review because of this move. Can you rebase it to have a separate commit which moves the code to the new file without functional changes, and then on top the commit that introduces new stuff? This way we can more reasonably review the interesting changes without reading everything in detail.

It'd be even better to move the commit that moves stuff to the very beginning: moving the stuff you just added doesn't make much sense.